Atomic/util.py

```diff
@@ -805,32 +803,32 @@ def load_scan_result_file(file_name):
 def file_lock(func):
     lock_file_name = "{}.lock".format(os.path.join(os.path.dirname(ATOMIC_INSTALL_JSON), "." + os.path.basename(ATOMIC_INSTALL_JSON)))
 
-    # Create the temporary lockfile if it doesn't exist
-    if not os.path.exists(lock_file_name):
-        open(lock_file_name, 'a').close()
-
-    install_data_file = open(lock_file_name, "r")
-
     def get_lock():
         '''
         Obtains a read-only file lock on the install data
         :return: 
         '''
         time_out = 0
         f_lock = False
-        while time_out < 10.5: # Ten second attempt to get a lock
-            try:
-                fcntl.flock(install_data_file, fcntl.LOCK_EX | fcntl.LOCK_NB)
-                f_lock = True
-                break
-            except IOError:
-                time.sleep(.5)
-                time_out += .5
+        # Create the temporary lockfile if it doesn't exist
+        if not os.path.exists(lock_file_name):
+            open(lock_file_name, 'a').close()
+
+        with open(lock_file_name, "r") as f:
+            while time_out < 10.5: # Ten second attempt to get a lock
+                try:
+                    fcntl.flock(f, fcntl.LOCK_EX | fcntl.LOCK_NB)
```

Comment:
Won't this lock be released when `f` is closed though?
Just tried this out:

```python
with open("lockfile", "r") as f:
    fcntl.flock(f, fcntl.LOCK_EX | fcntl.LOCK_NB)
with open("lockfile", "r") as f:
    fcntl.flock(f, fcntl.LOCK_EX | fcntl.LOCK_NB)
```

which works without issue. Whereas

```python
f = open("lockfile", "r")
fcntl.flock(f, fcntl.LOCK_EX | fcntl.LOCK_NB)

g = open("lockfile", "r")
fcntl.flock(g, fcntl.LOCK_EX | fcntl.LOCK_NB)
```

throws a BlockingIOError as expected.